app/helpers/append-query.js

```diff
@@ -0,0 +1,34 @@
+import { helper } from '@ember/component/helper';
+
+export function appendQuery([url = '', queryToAdd = '']/*, hash*/) {
+  if (typeof url !== 'string' || typeof queryToAdd !== 'string') {
+    return url;
+  }
+
+  // string any beginning ? characters
+  queryToAdd = queryToAdd.replace(/^\?/, '');
+
+  if (/\?$/.test(url)) {
+    // incoming url ends with a ?, just jam on the query string
+    return `${url}${queryToAdd}`;
+  }
+
+  // does a query string exist?
+  let [address, foundQuery] = url.split('?');
+
+  if (/\?/.test(address) || /\?/.test(foundQuery)) {
```

Comment:
If `address` & `foundQuery` are created when splitting against `?`, doesn’t that implicitly mean that neither `address` nor `foundQuery` will ever contain a `?` - so this condition will always be false?

It almost seems like a better weirdness check would be adding a third var to your destructure, i.e. `let [address, foundQuery, extra]` and then if somehow `extra` exists at all (meaning the url contains at least two `?`), just return early.

Reply:
yes, that is better! thanks